rhizome/host/lib/json_rpc_client.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require "json"
+require "socket"
+
+class JsonRpcError < StandardError
+  attr_reader :code
+
+  def initialize(message, code)
+    super(message)
+    @code = code
+  end
+end
+
+class JsonRpcClient
+  def initialize(socket_path, timeout = 5, response_size_limit = 1048576)
+    @socket_path = socket_path
+    @timeout = timeout
+    @response_size_limit = response_size_limit
+  end
+
+  def call(method, params = {})
+    # id is used to correlate the context between request and response.
+    # See https://www.jsonrpc.org/specification
+    id = rand(10000000)
+
+    payload = {
+      jsonrpc: "2.0",
+      method: method,
+      params: params,
+      id: id
+    }
+
+    unix_socket = UNIXSocket.new(@socket_path)
+    unix_socket.write_nonblock(payload.to_json + "\n")
+
+    response = JSON.parse(read_response(unix_socket))
+    if response["error"]
+      raise JsonRpcError.new(response["error"]["message"], response["error"]["code"])
+    end
```

Comment:
```suggestion
    if err = response["error"]
      raise JsonRpcError.new(err["message"], err["code"])
    end
```

Depending on your design theory, using `#fetch` if the program should crash instead of yielding nil upon non existence may be indicated.